lib/plugins/wallet/zcashd/zcashd.js

```diff
@@ -46,11 +46,42 @@ function balance (account, cryptoCode) {
 
 function sendCoins (account, address, cryptoAtoms, cryptoCode) {
   const coins = cryptoAtoms.shift(-unitScale).toFixed(8)
-
+  const checkSendStatus = function (opid) {
+    return new Promise((resolve, reject) => {
+      function checker(timeout) {
+        setTimeout(() => {
+          fetch('z_getoperationstatus', [[opid]])
+            .then(res => {
+              const status = _.get('status', res[0])
+              switch(status) {
+                case 'success': 
+                  resolve(res[0])
+                  break
+                case 'failed': 
+                  reject(res[0].error)
+                  break
+                case 'executing':
+                  if (timeout > 0) {
+                    checker(timeout - 500)
```

Comment:
Do we want a recursive call here? Could get messy.

Reply:
True, could get messy. Suggestions for writing a retry-until-time-limit thing in a promise chain without using recursion?